server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -783,10 +783,17 @@ private void checkCreateFlowInstancePermission(CreateFlowInstanceReq req) {
                 ConnectionConfig config = connectionService.getBasicWithoutPermissionCheck(req.getConnectionId());
                 parameters.getExportDbObjects().forEach(item -> {
                     if (item.getDbObjectType() == ObjectType.TABLE) {
-                        resource2Types.put(DBResource.from(config, req.getDatabaseName(), item.getObjectName()),
+                        resource2Types.put(
+                                DBResource.from(config, req.getDatabaseName(), item.getObjectName(),
+                                        ResourceType.ODC_TABLE),
                                 DatabasePermissionType.from(TaskType.EXPORT));
                     }
                 });
+            } else if (Objects.nonNull(parameters.isExportAllObjects()) && parameters.isExportAllObjects()) {
```

Comment:
exportAllObjects is a primitive field which can not be null~

Reply:
done